### ISSUE TITLE
Editing empty file crashes nvim, cc entire line causes error

### ISSUE THREAD

**Issue by quantum-booty** [nvim-treesitter/nvim-treesitter#479]
Editing an empty file causes nvim to crash, but editing a none empty file does not cause crashes. I've tested this on both .c and .py file, and  they both crashes.

![image](https://user-images.githubusercontent.com/28958399/93346492-f40d0e80-f82b-11ea-9e4b-45359a5c4a88.png)

If I edit a non-empty file, and do cc to change a line, it shows this error:
![image](https://user-images.githubusercontent.com/28958399/93346654-1d2d9f00-f82c-11ea-9d33-4d550c5da191.png)

**Output of `:checkhealth nvim_treesitter`**
Installation
  - OK: `git` executable found.
  - OK: `cc` executable found./

python parser healthcheck
  - OK: python parser found.
  - OK: `highlights.scm` found.
  - OK: `locals.scm` found.
  - OK: `textobjects.scm` found.
  - OK: `folds.scm` found.

**Output of `nvim --version`**
```
NVIM v0.5.0-695-gf34eeba2d
Build type: RelWithDebInfo
LuaJIT 2.1.0-beta3
Compilation: /usr/bin/gcc-5 -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=1 -O2 -g -Og -g -Wall -Wextra -pedantic -Wno-unused-parameter -Wstrict-prototypes -std=gnu99 -Wshadow -Wconversion -Wmissing-prototypes -Wvla -fstack-protector-strong -fno-common -fdiagnostics-color=auto -DINCLUDE_GENERATED_DECLARATIONS -D_GNU_SOURCE -DNVIM_MSGPACK_HAS_FLOAT32 -DNVIM_UNIBI_HAS_VAR_FROM -DMIN_LOG_LEVEL=3 -I/home/travis/build/neovim/bot-ci/build/neovim/build/config -I/home/travis/build/neovim/bot-ci/build/neovim/src -I/home/travis/build/neovim/bot-ci/build/neovim/.deps/usr/include -I/usr/include -I/home/travis/build/neovim/bot-ci/build/neovim/build/src/nvim/auto -I/home/travis/build/neovim/bot-ci/build/neovim/build/include
Compiled by travis@travis-job-3a429d31-c39e-4913-acf7-f93109be6de9

Features: +acl +iconv +tui
See ":help feature-compile"

   system vimrc file: "$VIM/sysinit.vim"
  fall-back for $VIM: "/share/nvim"

Run :checkhealth for more info
```

**init.vim file**
call plug#begin('~/.config/nvim/plugged')
Plug 'nvim-treesitter/nvim-treesitter'
call plug#end()

lua <<EOF
require'nvim-treesitter.configs'.setup {
  ensure_installed = "all",     -- one of "all", "language", or a list of languages

  highlight = {
    enable = true,
  },  
}
EOF

**Comment by vigoux** [comment]
Hi, thanks for reporting, but this is a duplicate of #440 and #423 .

**Comment by quantum-booty** [comment]
What about the cc error?

**Comment by vigoux** [comment]
The `cc` error is actually related to #423

**Comment by vigoux** [comment]
Little update on this, the `cc` error is actually a bug upstream, that I am working on at https://github.com/neovim/neovim/pull/12935, hopefully the fix will not take too long to land, I'll notify you here once it is done.

**Comment by quantum-booty** [comment]
Thank you so much!

**Comment by vigoux** [comment]
`cc`should be fixed on latest master now. Thank _you_ so much for reporting this issue !

**Comment by quantum-booty** [comment]
I just updated nvim to the latest release, and I am still having the cc error.
![image](https://user-images.githubusercontent.com/28958399/93668081-3e87c880-fa82-11ea-93e9-7adaeacc2e23.png)

**Comment by akinsho** [comment]
I've just tried this out on the latest nightly `NVIM v0.5.0-704-g905c2eb35` and I'm no longer getting the error on `cc` 🎉

**Comment by vigoux** [comment]
@quantum-booty can you prodive a full repro for this?
Thanks!

**Comment by quantum-booty** [comment]
@vigoux I am no longer getting the error, thank you so muny!
